extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote-init.ts

```diff
@@ -214,6 +221,32 @@ to pick-up automatically a free port`));
                     return;
                 }
 
+                // asked to send plugin resource
+                if (jsonParsed.internal.method === 'getResource') {
```

Comment:
Sorry, for stop method we have one more check:
`jsonParsed.internal.method && jsonParsed.internal.method === 'stop'`
Maybe you need to apply `jsonParsed.internal.method` to your code too? 
 => if (`jsonParsed.internal.methodjsonParsed.internal.method === 'getResource'`) {

Reply:
I don't think so. We are sure that `jsonParsed.internal` is present. Then we need to check `jsonParsed.internal.method` field. Of course, we can firstly check for undefined and then compare to method name, but I see no point because we may just check equality. If `method` field is undefined it is definitely not equal to `getResource`.